### DISCUSSION TITLE
Application memory usage reporting

### DISCUSSION THREAD

**Discussion by wwatu1337** [netdata/netdata#13867] [Q&A]
I have a question about memory reporting under applications. I have a python application that spawns 6 child processes(managed by supervisor),  grouped these processes together and netdata show about 600mb of usage, in reality they use just around 100mb together. By using top or htop, I can see why it would report 600mb, but in reality every child process uses only about 25mb as reported by smem, and by memory usage reporting in supervisor status, so I'm wondering is there any way to get it reporting correctly under applications, since under systemd services it reports correctly.

![image](https://user-images.githubusercontent.com/58936586/197395752-c2c32e71-854a-4894-b692-7ab1dc8f5074.png)
![image](https://user-images.githubusercontent.com/58936586/197395802-4d3686b5-1aaf-4005-bc91-cecae0bc08ac.png)
![image](https://user-images.githubusercontent.com/58936586/197395820-3505b2fe-cc4a-436b-b470-80b45270ea73.png)
![image](https://user-images.githubusercontent.com/58936586/197395843-49a6f81b-148f-415e-85d7-98166a1f00ad.png)

**Comment by Dim-P** [comment]
Hi @wwatu1337 and thank you for your post. 

The memory that is reported in the `apps.mem` chart is the difference between `VmRSS` (or what you see as `RSS` or `RES`) and the `shared` (`RssFile + RssShmem`) memory, which is effectively the `RssAnon` memory (size of resident anonymous memory) - see [proc(5)](https://man7.org/linux/man-pages/man5/proc.5.html) for more. I don't think what you have in your screenshot represents the `RssAnon` memory for your Python app. Which dimension have you selected in the chart?

Keep in mind that `apps.plugin` may group processes together, so you might be seeing the total memory consumption of a group of processes, in this instance. You can change that of course by editing the `apps_groups.conf` configuration.

**Comment by andrewm4894** [comment]
Thanks for the question @wwatu1337 apologies we were so slow in responding - i think it slipped through the cracks. 

Perhaps we could/should make a memory focused section in here: https://learn.netdata.cloud/docs/agent/collectors/apps.plugin#comparison-with-console-tools
